script.js

```diff
@@ -0,0 +1,841 @@
+      else keyDom.classList.remove("keyboard-line__button_pressed");
+    },
+    code: "ControlLeft",
+  },
+  {
+    type: "functional",
+    initial: "Win",
+    isClick: true,
+    func: (keyDom) => {
+      FUNCTIONAL_KEYS.windows = !FUNCTIONAL_KEYS.windows;
+      if (FUNCTIONAL_KEYS.windows) keyDom.classList.add("keyboard-line__button_pressed");
+      else keyDom.classList.remove("keyboard-line__button_pressed");
+    },
+    code: "MetaLeft",
+  },
+  {
+    type: "functional",
+    initial: "Alt",
+    isClick: true,
+    func: (keyDom) => {
+      FUNCTIONAL_KEYS.alt = !FUNCTIONAL_KEYS.alt;
+      if (FUNCTIONAL_KEYS.alt) keyDom.classList.add("keyboard-line__button_pressed");
+      else keyDom.classList.remove("keyboard-line__button_pressed");
+    },
+    code: "AltLeft",
+  },
+  {
+    type: "functional",
+    initial: "Space",
+    func: () => vKeyBoard.insertSymbol(" "),
+    code: "Space",
+  },
+  {
+    type: "functional",
+    initial: "Alt",
+    isClick: true,
+    func: (keyDom) => {
+      FUNCTIONAL_KEYS.altRight = !FUNCTIONAL_KEYS.altRight;
+      if (FUNCTIONAL_KEYS.altRight) keyDom.classList.add("keyboard-line__button_pressed");
+      else keyDom.classList.remove("keyboard-line__button_pressed");
+    },
+    code: "AltRight",
+  },
+  {
+    type: "functional",
+    initial: "←",
+    func() {
+      const cursorPositionStart = textArea.selectionStart;
+      vKeyBoard.focusTextArea(cursorPositionStart - 1);
+    },
+    code: "ArrowLeft",
+  },
+  {
+    type: "functional",
+    initial: "↓",
+    func() {
+      const cursorPosition = vKeyBoard.textArea.selectionStart;
+      const { value } = vKeyBoard.textArea;
+      const cursorLineBegin = value.substring(0, cursorPosition).lastIndexOf("\n") + 1;
+      const cursorLength = cursorPosition - cursorLineBegin;
+      let nextLineBegin = value.indexOf("\n", cursorPosition) + 1;
+      if (nextLineBegin < 1) nextLineBegin = value.length;
+      let nextLineLength = value.indexOf("\n", nextLineBegin) - nextLineBegin;
+      if (nextLineLength < 0) nextLineLength = value.length - nextLineBegin;
+      let newPosition = nextLineBegin
+        + (cursorLength > nextLineLength ? nextLineLength : cursorLength);
+      if (newPosition < 0) newPosition = value.length;
+      vKeyBoard.focusTextArea(newPosition);
+    },
+    code: "ArrowDown",
+  },
+  {
+    type: "functional",
+    initial: "→",
+    func() {
+      const cursorPositionStart = textArea.selectionStart;
+      vKeyBoard.focusTextArea(cursorPositionStart + 1);
+    },
+    code: "ArrowRight",
+  },
+  {
+    type: "functional",
+    initial: "Ctrl",
+    isClick: true,
+    func: (keyDom) => {
+      FUNCTIONAL_KEYS.ctrlRight = !FUNCTIONAL_KEYS.ctrlRight;
+      if (FUNCTIONAL_KEYS.ctrlRight) keyDom.classList.add("keyboard-line__button_pressed");
+      else keyDom.classList.remove("keyboard-line__button_pressed");
+    },
+    code: "ControlRight",
+  },
+]);
+
+function getType(key) {

```

Comment:
Functions, that just help to do smth is better to move to another file like helper.js or utils.js.

Reply:
Helper function was moved to another file.

https://github.com/BlackTruth/virtual-keyboard/blob/6297a0817c03affb1948c9e73a6352687e6ff449/src/helper.js#L1-L5